common/queue/src/main/java/org/thingsboard/server/queue/kafka/TbKafkaProducerTemplate.java

```diff
@@ -72,14 +79,33 @@ private TbKafkaProducerTemplate(TbKafkaSettings settings, String defaultTopic, S
     public void init() {
     }
 
+    void addAnalyticHeaders(List<Header> headers) {
+        try {
+            if (log.isDebugEnabled()) {
+                headers.add(new RecordHeader("_producerId", getClientId().getBytes(StandardCharsets.UTF_8)));
+                headers.add(new RecordHeader("_threadName", Thread.currentThread().getName().getBytes(StandardCharsets.UTF_8)));
+            }
+            if (log.isTraceEnabled()) {
```

Comment:
I suggest to add 
```
if (log.isDebugEnabled()) {
   addAnalyticHeaders(headers);
}
``` 

and remove isDebugEnabled from the addAnalyticHeaders. No need to check ```log.isTraceEnabled()``` if ```log.isDebugEnabled() == false```. Also, no need to put ```log.isDebugEnabled``` under ```try-catch``` block.

From the perf tests of MQTT broker I may note that log.isDebugEnabled is actually not that simple and requires some resources :-)